### ISSUE TITLE
kulla.jar error?

### ISSUE THREAD

**Issue by ygpark2** [Bachmann1234/java9_kernel#18]
Can you have a look my error?

---

TIMEOUT                                   Traceback (most recent call last)
/home/ygpark2/.pyenv/versions/3.5.0/lib/python3.5/site-packages/pexpect/expect.py in expect_loop(self=<pexpect.expect.Expecter object>, timeout=28.780590772628784)
     96                 # Still have time left, so read more data
---> 97                 incoming = spawn.read_nonblocking(spawn.maxread, timeout)
        incoming = 'jshell> '
        spawn.read_nonblocking = <bound method spawn.read_nonblocking of <pexpect.pty_spawn.spawn object at 0x7fe9d2945898>>
        spawn.maxread = 2000
        timeout = 28.780590772628784
     98                 time.sleep(0.0001)

/home/ygpark2/.pyenv/versions/3.5.0/lib/python3.5/site-packages/pexpect/pty_spawn.py in read_nonblocking(self=<pexpect.pty_spawn.spawn object>, size=2000, timeout=28.780590772628784)
    451             else:
--> 452                 raise TIMEOUT('Timeout exceeded.')
        global TIMEOUT = <class 'pexpect.exceptions.TIMEOUT'>
    453 

TIMEOUT: Timeout exceeded.

During handling of the above exception, another exception occurred:

TIMEOUT                                   Traceback (most recent call last)
/home/ygpark2/.pyenv/versions/3.5.0/lib/python3.5/runpy.py in _run_module_as_main(mod_name='__main__', alter_argv=0)
    168         sys.argv[0] = mod_spec.origin
    169     return _run_code(code, main_globals, None,
--> 170                      "__main__", mod_spec)
        mod_spec = ModuleSpec(name='**main**', loader=<_frozen_importlib_external.SourceFileLoader object at 0x7fe9da9be5c0>, origin='/home/ygpark2/.ipython/kernels/java/javakernel/__main__.py')
    171 
    172 def run_module(mod_name, init_globals=None,

/home/ygpark2/.pyenv/versions/3.5.0/lib/python3.5/runpy.py in _run_code(code=<code object <module> at 0x7fe9da9ad9c0, file "/...hon/kernels/java/javakernel/__main__.py", line 1>, run_globals={'IPKernelApp': <class 'ipykernel.kernelapp.IPKernelApp'>, 'JavaKernel': <class 'kernel.JavaKernel'>, '**builtins**': <module 'builtins' (built-in)>, '**cached**': '/home/ygpark2/.ipython/kernels/java/javakernel/**pycache**/**main**.cpython-35.pyc', '**doc**': None, '**file**': '/home/ygpark2/.ipython/kernels/java/javakernel/**main**.py', '**loader**': <_frozen_importlib_external.SourceFileLoader object>, '__name__': '**main**', '**package**': '', '**spec**': ModuleSpec(name='**main**', loader=<_frozen_impo...k2/.ipython/kernels/java/javakernel/__main__.py')}, init_globals=None, mod_name='**main**', mod_spec=ModuleSpec(name='**main**', loader=<_frozen_impo...k2/.ipython/kernels/java/javakernel/__main__.py'), pkg_name='', script_name=None)
     83                        **package** = pkg_name,
     84                        **spec** = mod_spec)
---> 85     exec(code, run_globals)
        global exec = undefined
        code = <code object <module> at 0x7fe9da9ad9c0, file "/home/ygpark2/.ipython/kernels/java/javakernel/**main**.py", line 1>
        run_globals = {'JavaKernel': <class 'kernel.JavaKernel'>, '**name**': '**main**', '**spec**': ModuleSpec(name='**main**', loader=<_frozen_importlib_external.SourceFileLoader object at 0x7fe9da9be5c0>, origin='/home/ygpark2/.ipython/kernels/java/javakernel/__main__.py'), '**cached**': '/home/ygpark2/.ipython/kernels/java/javakernel/**pycache**/**main**.cpython-35.pyc', 'IPKernelApp': <class 'ipykernel.kernelapp.IPKernelApp'>, '**package**': '', '**builtins**': <module 'builtins' (built-in)>, '**loader**': <_frozen_importlib_external.SourceFileLoader object at 0x7fe9da9be5c0>, '__doc__': None, '**file**': '/home/ygpark2/.ipython/kernels/java/javakernel/**main**.py'}
     86     return run_globals
     87 

/home/ygpark2/.ipython/kernels/java/javakernel/**main**.py in <module>()
      7     except:
      8         from IPython.kernel.zmq.kernelapp import IPKernelApp
----> 9     IPKernelApp.launch_instance(kernel_class=JavaKernel)
        global IPKernelApp.launch_instance = <bound method Application.launch_instance of <class 'ipykernel.kernelapp.IPKernelApp'>>
        global kernel_class = undefined
        global JavaKernel = <class 'kernel.JavaKernel'>

/home/ygpark2/.pyenv/versions/3.5.0/lib/python3.5/site-packages/traitlets/config/application.py in launch_instance(cls=<class 'ipykernel.kernelapp.IPKernelApp'>, argv=None, **kwargs={'kernel_class': <class 'kernel.JavaKernel'>})
    589         """
    590         app = cls.instance(**kwargs)
--> 591         app.initialize(argv)
        app.initialize = <bound method IPKernelApp.initialize of <ipykernel.kernelapp.IPKernelApp object at 0x7fe9da9be550>>
        argv = None
    592         app.start()
    593 

<decorator-gen-121> in initialize(self=<ipykernel.kernelapp.IPKernelApp object>, argv=None)

/home/ygpark2/.pyenv/versions/3.5.0/lib/python3.5/site-packages/traitlets/config/application.py in catch_config_error(method=<function IPKernelApp.initialize>, app=<ipykernel.kernelapp.IPKernelApp object>, _args=(None,), *_kwargs={})
     73     """
     74     try:
---> 75         return method(app, _args, *_kwargs)
        method = <function IPKernelApp.initialize at 0x7fe9d29b7510>
        app = <ipykernel.kernelapp.IPKernelApp object at 0x7fe9da9be550>
        args = (None,)
        kwargs = {}
     76     except (TraitError, ArgumentError) as e:
     77         app.print_help()

/home/ygpark2/.pyenv/versions/3.5.0/lib/python3.5/site-packages/ipykernel/kernelapp.py in initialize(self=<ipykernel.kernelapp.IPKernelApp object>, argv=None)
    380         self.init_io()
    381         self.init_signal()
--> 382         self.init_kernel()
        self.init_kernel = <bound method IPKernelApp.init_kernel of <ipykernel.kernelapp.IPKernelApp object at 0x7fe9da9be550>>
    383         # shell init steps
    384         self.init_path()

/home/ygpark2/.pyenv/versions/3.5.0/lib/python3.5/site-packages/ipykernel/kernelapp.py in init_kernel(self=<ipykernel.kernelapp.IPKernelApp object>)
    319                                 log=self.log,
    320                                 profile_dir=self.profile_dir,
--> 321                                 user_ns=self.user_ns,
        global user_ns = undefined
        self.user_ns = None
    322         )
    323         kernel.record_ports(self.ports)

/home/ygpark2/.pyenv/versions/3.5.0/lib/python3.5/site-packages/traitlets/config/configurable.py in instance(cls=<class 'kernel.JavaKernel'>, _args=(), *_kwargs={'iopub_socket': <zmq.sugar.socket.Socket object>, 'log': <logging.Logger object>, 'parent': <ipykernel.kernelapp.IPKernelApp object>, 'profile_dir': <IPython.core.profiledir.ProfileDir object>, 'session': <jupyter_client.session.Session object>, 'shell_streams': [<zmq.eventloop.zmqstream.ZMQStream object>, <zmq.eventloop.zmqstream.ZMQStream object>], 'stdin_socket': <zmq.sugar.socket.Socket object>, 'user_ns': None})
    378         # Create and save the instance
    379         if cls._instance is None:
--> 380             inst = cls(_args, *_kwargs)
        inst = undefined
        cls = <class 'kernel.JavaKernel'>
        args = ()
        kwargs = {'stdin_socket': <zmq.sugar.socket.Socket object at 0x7fe9d2956048>, 'iopub_socket': <zmq.sugar.socket.Socket object at 0x7fe9d2998f48>, 'profile_dir': <IPython.core.profiledir.ProfileDir object at 0x7fe9d2942e80>, 'parent': <ipykernel.kernelapp.IPKernelApp object at 0x7fe9da9be550>, 'shell_streams': [<zmq.eventloop.zmqstream.ZMQStream object at 0x7fe9d29b66a0>, <zmq.eventloop.zmqstream.ZMQStream object at 0x7fe9d29b6240>], 'log': <logging.Logger object at 0x7fe9d938c710>, 'user_ns': None, 'session': <jupyter_client.session.Session object at 0x7fe9d2942a20>}
    381             # Now make sure that the instance will also be returned by
    382             # parent classes' _instance attribute.

/home/ygpark2/.ipython/kernels/java/javakernel/kernel.py in **init**(self=<kernel.JavaKernel object>, **kwargs={'iopub_socket': <zmq.sugar.socket.Socket object>, 'log': <logging.Logger object>, 'parent': <ipykernel.kernelapp.IPKernelApp object>, 'profile_dir': <IPython.core.profiledir.ProfileDir object>, 'session': <jupyter_client.session.Session object>, 'shell_streams': [<zmq.eventloop.zmqstream.ZMQStream object>, <zmq.eventloop.zmqstream.ZMQStream object>], 'stdin_socket': <zmq.sugar.socket.Socket object>, 'user_ns': None})
     32         self.env = {"JAVA_9_HOME": os.environ['JAVA_9_HOME'],
     33                     "KULLA_HOME": os.environ['KULLA_HOME']}
---> 34         self._start_java_repl()
        self._start_java_repl = <bound method JavaKernel._start_java_repl of <kernel.JavaKernel object at 0x7fe9d2945978>>
     35 
     36     @property

/home/ygpark2/.ipython/kernels/java/javakernel/kernel.py in _start_java_repl(self=<kernel.JavaKernel object>)
     50                 u'->',
     51                 None,
---> 52                 continuation_prompt=u'>>'
        global continuation_prompt = undefined
     53             )
     54         finally:

/home/ygpark2/.pyenv/versions/3.5.0/lib/python3.5/site-packages/pexpect/replwrap.py in **init**(self=<pexpect.replwrap.REPLWrapper object>, cmd_or_spawn='/home/ygpark2/prog/jdk-9/bin/java -jar /home/ygpark2/prog/kulla.jar', orig_prompt='->', prompt_change=None, new_prompt='[PEXPECT_PROMPT>', continuation_prompt='>>', extra_init_cmd=None)
     53         self.continuation_prompt = continuation_prompt
     54 
---> 55         self._expect_prompt()
        self._expect_prompt = <bound method REPLWrapper._expect_prompt of <pexpect.replwrap.REPLWrapper object at 0x7fe9d2942e48>>
     56 
     57         if extra_init_cmd is not None:

/home/ygpark2/.pyenv/versions/3.5.0/lib/python3.5/site-packages/pexpect/replwrap.py in _expect_prompt(self=<pexpect.replwrap.REPLWrapper object>, timeout=-1)
     64     def _expect_prompt(self, timeout=-1):
     65         return self.child.expect_exact([self.prompt, self.continuation_prompt],
---> 66                                        timeout=timeout)
        timeout = -1
     67 
     68     def run_command(self, command, timeout=-1):

/home/ygpark2/.pyenv/versions/3.5.0/lib/python3.5/site-packages/pexpect/spawnbase.py in expect_exact(self=<pexpect.pty_spawn.spawn object>, pattern_list=['->', '>>'], timeout=30, searchwindowsize=-1, async=False)
    382             return expect_async(exp, timeout)
    383         else:
--> 384             return exp.expect_loop(timeout)
        exp.expect_loop = <bound method Expecter.expect_loop of <pexpect.expect.Expecter object at 0x7fe9d2945630>>
        timeout = 30
    385 
    386     def expect_loop(self, searcher, timeout=-1, searchwindowsize=-1):

/home/ygpark2/.pyenv/versions/3.5.0/lib/python3.5/site-packages/pexpect/expect.py in expect_loop(self=<pexpect.expect.Expecter object>, timeout=28.780590772628784)
    102             return self.eof(e)
    103         except TIMEOUT as e:
--> 104             return self.timeout(e)
        self.timeout = <bound method Expecter.timeout of <pexpect.expect.Expecter object at 0x7fe9d2945630>>
        e = undefined
    105         except:
    106             self.errored()

/home/ygpark2/.pyenv/versions/3.5.0/lib/python3.5/site-packages/pexpect/expect.py in timeout(self=<pexpect.expect.Expecter object>, err=TIMEOUT('Timeout exceeded.',))
     66             if err is not None:
     67                 msg = str(err) + '\n' + msg
---> 68             raise TIMEOUT(msg)
        TIMEOUT = <class 'pexpect.exceptions.TIMEOUT'>
        msg = "Timeout exceeded.\n<pexpect.pty_spawn.spawn object at 0x7fe9d2945898>\ncommand: /home/ygpark2/prog/jdk-9/bin/java\nargs: ['/home/ygpark2/prog/jdk-9/bin/java', '-jar', '/home/ygpark2/prog/kulla.jar']\nsearcher: None\nbuffer (last 100 chars): '|  Welcome to JShell -- Version 9-ea\r\n|  For an introduction type: /help intro\r\n\r\n\r\njshell> '\nbefore (last 100 chars): '|  Welcome to JShell -- Version 9-ea\r\n|  For an introduction type: /help intro\r\n\r\n\r\njshell> '\nafter: <class 'pexpect.exceptions.TIMEOUT'>\nmatch: None\nmatch_index: None\nexitstatus: None\nflag_eof: False\npid: 4077\nchild_fd: 34\nclosed: False\ntimeout: 30\ndelimiter: <class 'pexpect.exceptions.EOF'>\nlogfile: None\nlogfile_read: None\nlogfile_send: None\nmaxread: 2000\nignorecase: False\nsearchwindowsize: None\ndelaybeforesend: 0.05\ndelayafterclose: 0.1\ndelayafterterminate: 0.1"
     69 
     70     def errored(self):

TIMEOUT: Timeout exceeded.
<pexpect.pty_spawn.spawn object at 0x7fe9d2945898>
command: /home/ygpark2/prog/jdk-9/bin/java
args: ['/home/ygpark2/prog/jdk-9/bin/java', '-jar', '/home/ygpark2/prog/kulla.jar']
searcher: None
buffer (last 100 chars): '|  Welcome to JShell -- Version 9-ea\r\n|  For an introduction type: /help intro\r\n\r\n\r\njshell> '
before (last 100 chars): '|  Welcome to JShell -- Version 9-ea\r\n|  For an introduction type: /help intro\r\n\r\n\r\njshell> '
after: <class 'pexpect.exceptions.TIMEOUT'>
match: None
match_index: None
exitstatus: None
flag_eof: False
pid: 4077
child_fd: 34
closed: False
timeout: 30
delimiter: <class 'pexpect.exceptions.EOF'>
logfile: None
logfile_read: None
logfile_send: None
maxread: 2000
ignorecase: False
searchwindowsize: None
delaybeforesend: 0.05
delayafterclose: 0.1
delayafterterminate: 0.1
[I 13:43:48.073 NotebookApp] KernelRestarter: restarting kernel (1/5)
WARNING:root:kernel 74ba1dd4-1095-4b10-b7bd-15c082076de0 restarted

---

TIMEOUT                                   Traceback (most recent call last)
/home/ygpark2/.pyenv/versions/3.5.0/lib/python3.5/site-packages/pexpect/expect.py in expect_loop(self=<pexpect.expect.Expecter object>, timeout=28.63274836540222)
     96                 # Still have time left, so read more data
---> 97                 incoming = spawn.read_nonblocking(spawn.maxread, timeout)
        incoming = '\r\n\r\njshell> '
        spawn.read_nonblocking = <bound method spawn.read_nonblocking of <pexpect.pty_spawn.spawn object at 0x7f651026f7b8>>
        spawn.maxread = 2000
        timeout = 28.63274836540222
     98                 time.sleep(0.0001)

/home/ygpark2/.pyenv/versions/3.5.0/lib/python3.5/site-packages/pexpect/pty_spawn.py in read_nonblocking(self=<pexpect.pty_spawn.spawn object>, size=2000, timeout=28.63274836540222)
    451             else:
--> 452                 raise TIMEOUT('Timeout exceeded.')
        global TIMEOUT = <class 'pexpect.exceptions.TIMEOUT'>
    453 

TIMEOUT: Timeout exceeded.

During handling of the above exception, another exception occurred:

TIMEOUT                                   Traceback (most recent call last)
/home/ygpark2/.pyenv/versions/3.5.0/lib/python3.5/runpy.py in _run_module_as_main(mod_name='__main__', alter_argv=0)
    168         sys.argv[0] = mod_spec.origin
    169     return _run_code(code, main_globals, None,
--> 170                      "__main__", mod_spec)
        mod_spec = ModuleSpec(name='**main**', loader=<_frozen_importlib_external.SourceFileLoader object at 0x7f65182e85c0>, origin='/home/ygpark2/.ipython/kernels/java/javakernel/__main__.py')
    171 
    172 def run_module(mod_name, init_globals=None,

/home/ygpark2/.pyenv/versions/3.5.0/lib/python3.5/runpy.py in _run_code(code=<code object <module> at 0x7f65182d79c0, file "/...hon/kernels/java/javakernel/__main__.py", line 1>, run_globals={'IPKernelApp': <class 'ipykernel.kernelapp.IPKernelApp'>, 'JavaKernel': <class 'kernel.JavaKernel'>, '**builtins**': <module 'builtins' (built-in)>, '**cached**': '/home/ygpark2/.ipython/kernels/java/javakernel/**pycache**/**main**.cpython-35.pyc', '**doc**': None, '**file**': '/home/ygpark2/.ipython/kernels/java/javakernel/**main**.py', '**loader**': <_frozen_importlib_external.SourceFileLoader object>, '__name__': '**main**', '**package**': '', '**spec**': ModuleSpec(name='**main**', loader=<_frozen_impo...k2/.ipython/kernels/java/javakernel/__main__.py')}, init_globals=None, mod_name='**main**', mod_spec=ModuleSpec(name='**main**', loader=<_frozen_impo...k2/.ipython/kernels/java/javakernel/__main__.py'), pkg_name='', script_name=None)
     83                        **package** = pkg_name,
     84                        **spec** = mod_spec)
---> 85     exec(code, run_globals)
        global exec = undefined
        code = <code object <module> at 0x7f65182d79c0, file "/home/ygpark2/.ipython/kernels/java/javakernel/**main**.py", line 1>
        run_globals = {'**file**': '/home/ygpark2/.ipython/kernels/java/javakernel/**main**.py', '**loader**': <_frozen_importlib_external.SourceFileLoader object at 0x7f65182e85c0>, 'IPKernelApp': <class 'ipykernel.kernelapp.IPKernelApp'>, 'JavaKernel': <class 'kernel.JavaKernel'>, '__package__': '', '**spec**': ModuleSpec(name='**main**', loader=<_frozen_importlib_external.SourceFileLoader object at 0x7f65182e85c0>, origin='/home/ygpark2/.ipython/kernels/java/javakernel/__main__.py'), '**name**': '**main**', '**builtins**': <module 'builtins' (built-in)>, '**cached**': '/home/ygpark2/.ipython/kernels/java/javakernel/**pycache**/**main**.cpython-35.pyc', '**doc**': None}
     86     return run_globals
     87 

/home/ygpark2/.ipython/kernels/java/javakernel/**main**.py in <module>()
      7     except:
      8         from IPython.kernel.zmq.kernelapp import IPKernelApp
----> 9     IPKernelApp.launch_instance(kernel_class=JavaKernel)
        global IPKernelApp.launch_instance = <bound method Application.launch_instance of <class 'ipykernel.kernelapp.IPKernelApp'>>
        global kernel_class = undefined
        global JavaKernel = <class 'kernel.JavaKernel'>

/home/ygpark2/.pyenv/versions/3.5.0/lib/python3.5/site-packages/traitlets/config/application.py in launch_instance(cls=<class 'ipykernel.kernelapp.IPKernelApp'>, argv=None, **kwargs={'kernel_class': <class 'kernel.JavaKernel'>})
    589         """
    590         app = cls.instance(**kwargs)
--> 591         app.initialize(argv)
        app.initialize = <bound method IPKernelApp.initialize of <ipykernel.kernelapp.IPKernelApp object at 0x7f65182e8550>>
        argv = None
    592         app.start()
    593 

<decorator-gen-121> in initialize(self=<ipykernel.kernelapp.IPKernelApp object>, argv=None)

/home/ygpark2/.pyenv/versions/3.5.0/lib/python3.5/site-packages/traitlets/config/application.py in catch_config_error(method=<function IPKernelApp.initialize>, app=<ipykernel.kernelapp.IPKernelApp object>, _args=(None,), *_kwargs={})
     73     """
     74     try:
---> 75         return method(app, _args, *_kwargs)
        method = <function IPKernelApp.initialize at 0x7f65102e1510>
        app = <ipykernel.kernelapp.IPKernelApp object at 0x7f65182e8550>
        args = (None,)
        kwargs = {}
     76     except (TraitError, ArgumentError) as e:
     77         app.print_help()

/home/ygpark2/.pyenv/versions/3.5.0/lib/python3.5/site-packages/ipykernel/kernelapp.py in initialize(self=<ipykernel.kernelapp.IPKernelApp object>, argv=None)
    380         self.init_io()
    381         self.init_signal()
--> 382         self.init_kernel()
        self.init_kernel = <bound method IPKernelApp.init_kernel of <ipykernel.kernelapp.IPKernelApp object at 0x7f65182e8550>>
    383         # shell init steps
    384         self.init_path()

/home/ygpark2/.pyenv/versions/3.5.0/lib/python3.5/site-packages/ipykernel/kernelapp.py in init_kernel(self=<ipykernel.kernelapp.IPKernelApp object>)
    319                                 log=self.log,
    320                                 profile_dir=self.profile_dir,
--> 321                                 user_ns=self.user_ns,
        global user_ns = undefined
        self.user_ns = None
    322         )
    323         kernel.record_ports(self.ports)

/home/ygpark2/.pyenv/versions/3.5.0/lib/python3.5/site-packages/traitlets/config/configurable.py in instance(cls=<class 'kernel.JavaKernel'>, _args=(), *_kwargs={'iopub_socket': <zmq.sugar.socket.Socket object>, 'log': <logging.Logger object>, 'parent': <ipykernel.kernelapp.IPKernelApp object>, 'profile_dir': <IPython.core.profiledir.ProfileDir object>, 'session': <jupyter_client.session.Session object>, 'shell_streams': [<zmq.eventloop.zmqstream.ZMQStream object>, <zmq.eventloop.zmqstream.ZMQStream object>], 'stdin_socket': <zmq.sugar.socket.Socket object>, 'user_ns': None})
    378         # Create and save the instance
    379         if cls._instance is None:
--> 380             inst = cls(_args, *_kwargs)
        inst = undefined
        cls = <class 'kernel.JavaKernel'>
        args = ()
        kwargs = {'stdin_socket': <zmq.sugar.socket.Socket object at 0x7f6510280048>, 'profile_dir': <IPython.core.profiledir.ProfileDir object at 0x7f651026ce80>, 'user_ns': None, 'iopub_socket': <zmq.sugar.socket.Socket object at 0x7f65102c2f48>, 'shell_streams': [<zmq.eventloop.zmqstream.ZMQStream object at 0x7f65102e0710>, <zmq.eventloop.zmqstream.ZMQStream object at 0x7f65102e0240>], 'log': <logging.Logger object at 0x7f6516cb6710>, 'parent': <ipykernel.kernelapp.IPKernelApp object at 0x7f65182e8550>, 'session': <jupyter_client.session.Session object at 0x7f651026ca20>}
    381             # Now make sure that the instance will also be returned by
    382             # parent classes' _instance attribute.

/home/ygpark2/.ipython/kernels/java/javakernel/kernel.py in **init**(self=<kernel.JavaKernel object>, **kwargs={'iopub_socket': <zmq.sugar.socket.Socket object>, 'log': <logging.Logger object>, 'parent': <ipykernel.kernelapp.IPKernelApp object>, 'profile_dir': <IPython.core.profiledir.ProfileDir object>, 'session': <jupyter_client.session.Session object>, 'shell_streams': [<zmq.eventloop.zmqstream.ZMQStream object>, <zmq.eventloop.zmqstream.ZMQStream object>], 'stdin_socket': <zmq.sugar.socket.Socket object>, 'user_ns': None})
     32         self.env = {"JAVA_9_HOME": os.environ['JAVA_9_HOME'],
     33                     "KULLA_HOME": os.environ['KULLA_HOME']}
---> 34         self._start_java_repl()
        self._start_java_repl = <bound method JavaKernel._start_java_repl of <kernel.JavaKernel object at 0x7f651026f978>>
     35 
     36     @property

/home/ygpark2/.ipython/kernels/java/javakernel/kernel.py in _start_java_repl(self=<kernel.JavaKernel object>)
     50                 u'->',
     51                 None,
---> 52                 continuation_prompt=u'>>'
        global continuation_prompt = undefined
     53             )
     54         finally:

/home/ygpark2/.pyenv/versions/3.5.0/lib/python3.5/site-packages/pexpect/replwrap.py in **init**(self=<pexpect.replwrap.REPLWrapper object>, cmd_or_spawn='/home/ygpark2/prog/jdk-9/bin/java -jar /home/ygpark2/prog/kulla.jar', orig_prompt='->', prompt_change=None, new_prompt='[PEXPECT_PROMPT>', continuation_prompt='>>', extra_init_cmd=None)
     53         self.continuation_prompt = continuation_prompt
     54 
---> 55         self._expect_prompt()
        self._expect_prompt = <bound method REPLWrapper._expect_prompt of <pexpect.replwrap.REPLWrapper object at 0x7f651026f9b0>>
     56 
     57         if extra_init_cmd is not None:

/home/ygpark2/.pyenv/versions/3.5.0/lib/python3.5/site-packages/pexpect/replwrap.py in _expect_prompt(self=<pexpect.replwrap.REPLWrapper object>, timeout=-1)
     64     def _expect_prompt(self, timeout=-1):
     65         return self.child.expect_exact([self.prompt, self.continuation_prompt],
---> 66                                        timeout=timeout)
        timeout = -1
     67 
     68     def run_command(self, command, timeout=-1):

/home/ygpark2/.pyenv/versions/3.5.0/lib/python3.5/site-packages/pexpect/spawnbase.py in expect_exact(self=<pexpect.pty_spawn.spawn object>, pattern_list=['->', '>>'], timeout=30, searchwindowsize=-1, async=False)
    382             return expect_async(exp, timeout)
    383         else:
--> 384             return exp.expect_loop(timeout)
        exp.expect_loop = <bound method Expecter.expect_loop of <pexpect.expect.Expecter object at 0x7f651026f668>>
        timeout = 30
    385 
    386     def expect_loop(self, searcher, timeout=-1, searchwindowsize=-1):

/home/ygpark2/.pyenv/versions/3.5.0/lib/python3.5/site-packages/pexpect/expect.py in expect_loop(self=<pexpect.expect.Expecter object>, timeout=28.63274836540222)
    102             return self.eof(e)
    103         except TIMEOUT as e:
--> 104             return self.timeout(e)
        self.timeout = <bound method Expecter.timeout of <pexpect.expect.Expecter object at 0x7f651026f668>>
        e = undefined
    105         except:
    106             self.errored()

/home/ygpark2/.pyenv/versions/3.5.0/lib/python3.5/site-packages/pexpect/expect.py in timeout(self=<pexpect.expect.Expecter object>, err=TIMEOUT('Timeout exceeded.',))
     66             if err is not None:
     67                 msg = str(err) + '\n' + msg
---> 68             raise TIMEOUT(msg)
        TIMEOUT = <class 'pexpect.exceptions.TIMEOUT'>
        msg = "Timeout exceeded.\n<pexpect.pty_spawn.spawn object at 0x7f651026f7b8>\ncommand: /home/ygpark2/prog/jdk-9/bin/java\nargs: ['/home/ygpark2/prog/jdk-9/bin/java', '-jar', '/home/ygpark2/prog/kulla.jar']\nsearcher: None\nbuffer (last 100 chars): '|  Welcome to JShell -- Version 9-ea\r\n|  For an introduction type: /help intro\r\n\r\n\r\njshell> '\nbefore (last 100 chars): '|  Welcome to JShell -- Version 9-ea\r\n|  For an introduction type: /help intro\r\n\r\n\r\njshell> '\nafter: <class 'pexpect.exceptions.TIMEOUT'>\nmatch: None\nmatch_index: None\nexitstatus: None\nflag_eof: False\npid: 4231\nchild_fd: 34\nclosed: False\ntimeout: 30\ndelimiter: <class 'pexpect.exceptions.EOF'>\nlogfile: None\nlogfile_read: None\nlogfile_send: None\nmaxread: 2000\nignorecase: False\nsearchwindowsize: None\ndelaybeforesend: 0.05\ndelayafterclose: 0.1\ndelayafterterminate: 0.1"
     69 
     70     def errored(self):

TIMEOUT: Timeout exceeded.
<pexpect.pty_spawn.spawn object at 0x7f651026f7b8>
command: /home/ygpark2/prog/jdk-9/bin/java
args: ['/home/ygpark2/prog/jdk-9/bin/java', '-jar', '/home/ygpark2/prog/kulla.jar']
searcher: None
buffer (last 100 chars): '|  Welcome to JShell -- Version 9-ea\r\n|  For an introduction type: /help intro\r\n\r\n\r\njshell> '
before (last 100 chars): '|  Welcome to JShell -- Version 9-ea\r\n|  For an introduction type: /help intro\r\n\r\n\r\njshell> '
after: <class 'pexpect.exceptions.TIMEOUT'>
match: None
match_index: None
exitstatus: None
flag_eof: False
pid: 4231
child_fd: 34
closed: False
timeout: 30
delimiter: <class 'pexpect.exceptions.EOF'>
logfile: None
logfile_read: None
logfile_send: None
maxread: 2000
ignorecase: False
searchwindowsize: None
delaybeforesend: 0.05
delayafterclose: 0.1
delayafterterminate: 0.1

**Comment by Bachmann1234** [comment]
Is this trying to start up the kernet? Or running some java?

Best I can tell you is that some command is taking too long and ipython is exploding because of it.

**Comment by ygpark2** [comment]
@Bachmann1234 

![2016-04-28 10 06 42](https://cloud.githubusercontent.com/assets/93011/14872638/30ecb0dc-0d29-11e6-838d-3088e4f7ad44.png)

After waiting a while, I can see the next screen.

![2016-04-28 10 07 07](https://cloud.githubusercontent.com/assets/93011/14872637/30e82d64-0d29-11e6-9b13-aa464dd379d5.png)

It seems like happening on starting up the kernel.

**Comment by Bachmann1234** [comment]
My best guess is that Kulla jar is not starting. Can you run the Java 9 repel jar without IPython?

**Comment by ygpark2** [comment]
![2016-04-28 13 10 40](https://cloud.githubusercontent.com/assets/93011/14875164/a16eaad6-0d42-11e6-83f8-2ae57b3375f6.png)

Is this what you want it?
Let me know if you want me to try something different?

**Comment by Bachmann1234** [comment]
Sorry man, I have no idea. The kernel is having trouble talking to Kulla. I had thought maybe something is wrong with kulla but if it runs then im back to not knowing.

**Comment by Bachmann1234** [comment]
Hey, I managed to fix this but I am hitting another issue. 

Basically the underlying kulla.jar had some changes that broke the kernel. I tried to update it. 

The kernel should now load but multiline code does not seem to be working despite my efforts :-/

I will open a new issue for that
